microcloud/test/suites/basic.sh

```diff
@@ -822,3 +822,135 @@ test_auto() {
+  echo Create a MicroCloud that re-uses an existing MicroCeph and MicroOVN on different nodes
+  lxc exec micro02 -- microceph cluster bootstrap
+  lxc exec micro03 -- microovn cluster bootstrap
+  microcloud_interactive | lxc exec micro01 -- sh -c "microcloud init > out"
+  services_validator
+
+  reset_systems 3 3 3
+  echo Create a MicroCloud that re-uses an existing existing service with multiple nodes from this cluster
+  export REUSE_EXISTING_COUNT=1
+  export REUSE_EXISTING="add"
+  lxc exec micro02 -- microceph cluster bootstrap
+  token="$(lxc exec micro02 -- microceph cluster add micro01)"
+  lxc exec micro01 -- microceph cluster join "${token}"
+  microcloud_interactive | lxc exec micro01 -- sh -c "microcloud init > out"
+  services_validator
+
+  reset_systems 3 3 3
+  echo Create a MicroCloud that re-uses an existing existing service with all nodes from this cluster
+  lxc exec micro02 -- microceph cluster bootstrap
+  token="$(lxc exec micro02 -- microceph cluster add micro01)"
+  lxc exec micro01 -- microceph cluster join "${token}"
+  token="$(lxc exec micro02 -- microceph cluster add micro03)"
+  lxc exec micro03 -- microceph cluster join "${token}"
+  microcloud_interactive | lxc exec micro01 -- sh -c "microcloud init > out"
+  services_validator
+
+  reset_systems 4 3 3
+  echo Create a MicroCloud that re-uses an existing existing service with foreign cluster members
+  lxc exec micro04 -- snap disable microcloud
+  lxc exec micro02 -- microceph cluster bootstrap
+  token="$(lxc exec micro02 -- microceph cluster add micro04)"
+  lxc exec micro04 -- microceph cluster join "${token}"
+  microcloud_interactive | lxc exec micro01 -- sh -c "microcloud init > out"
+  services_validator
+  validate_system_microceph micro04 1
+
+  reset_systems 3 3 3
+  echo Fail to create a MicroCloud due to an existing service if --auto specified
+  lxc exec micro02 -- microceph cluster bootstrap
+  ! lxc exec micro01 -- sh -c "TEST_CONSOLE=0 microcloud init --auto > out" || true
+
+
+
+  echo Fail to create a MicroCloud due to conflicting existing services
+  lxc exec micro03 -- microceph cluster bootstrap
+  ! microcloud_interactive | lxc exec micro01 -- sh -c "microcloud init > out" || true
+
+  reset_systems 3 3 3
+  echo Create a MicroCloud that re-uses an existing service with preseed
+  addr=$(lxc ls micro01 -f csv -c4 | grep enp5s0 | cut -d' ' -f1)
+  lxc exec micro01 -- sh -c "
+  cat << EOF > /root/preseed.yaml
+lookup_subnet: ${addr}/24
+lookup_interface: enp5s0
+reuse_existing_clusters: true
+systems:
+- name: micro01
+- name: micro02
+- name: micro03
+ovn:
+  ipv4_gateway: 10.1.123.1/24
+  ipv4_range: 10.1.123.100-10.1.123.254
+  ipv6_gateway: fd42:1:1234:1234::1/64
+  dns_servers: 10.1.123.1,8.8.8.8
+storage:
+  local:
+    - find: id == sdb
+      wipe: true
+  ceph:
+    - find: id == sdc
+      wipe: true
+  cephfs: true
+"
+  lxc exec micro01 -- sh -c "cat /root/preseed.yaml | TEST_CONSOLE=0 microcloud init --preseed"
```

Comment:
Is the intermediate `/root/preseed.yaml` needed/useful for debugging? If not, would it be possible to use something like:
```
lxc exec micro01 --env TEST_CONSOLE=0 -- microcloud init --preseed << EOF

lookup_subnet: ${addr}/24
lookup_interface: enp5s0
reuse_existing_clusters: true
systems:
...
EOF
```